pkg/model/firewall.go

```diff
@@ -196,7 +196,7 @@ func (b *FirewallModelBuilder) applyNodeToMasterBlockSpecificPorts(c *fi.ModelBu
 	udpRanges := []portRange{{From: 1, To: 65535}}
 	protocols := []Protocol{}
 
-	if b.Cluster.Spec.Networking.Calico != nil {
+	if b.Cluster.Spec.Networking.Calico != nil && b.Cluster.Spec.Networking.Calico.KubernetesDataStore != true {
```

Comment:
Nit: `!x` is probably more idiomatic than `x != true`, but I actually find your version easier to read.  So either works :-)